sips/sip-013/sip-013-semi-fungible-token-standard.md

```diff
@@ -0,0 +1,232 @@
+
+Transfer many tokens in one contract call and emit a memo for each. This function follows the same procedure as `transfer-many` but will emit the memo contained in the tuple after each transfer. The whole function call should fail with an `err` response if one of the transfers fails.
+
+## Trait implementation
+
+An implementation of the proposed trait is provided below.
+
+```clarity
+(define-trait sip013-semi-fungible-token-trait
+	(
+		;; Get a token type balance of the passed principal.
+		(get-balance (uint principal) (response uint uint))
+
+		;; Get the total SFT balance of the passed principal.
+		(get-overall-balance (principal) (response uint uint))
+
+		;; Get the current total supply of a token type.
+		(get-total-supply (uint) (response uint uint))
+
+		;; Get the overall SFT supply.
+		(get-overall-supply () (response uint uint))
+
+		;; Get the number of decimal places of a token type.
+		(get-decimals (uint) (response uint uint))
+
+		;; Get an optional token URI that represents metadata for a specific token.
+		(get-token-uri (uint) (response (optional (string-utf8 256)) uint))
+
+		;; Transfer from one principal to another.
+		(transfer (uint uint principal principal) (response bool uint))
+
+		;; Transfer from one principal to another with a memo.
+		(transfer-memo (uint uint principal principal (buff 34)) (response bool uint))
+
+		;; Transfer many tokens at once.
+		(transfer-many ((list 200 {token-id: uint, amount: uint, sender: principal, recipient: principal})) (response bool uint))
+
+		;; Transfer many tokens at once with memos.
+		(transfer-many-memo ((list 200 {token-id: uint, amount: uint, sender: principal, recipient: principal, memo: (buff 34)})) (response bool uint))
+	)
+)
+```
+## Events
+
+Semi-fungible token contracts should emit custom events in certain situations via `print`. These events should be emitted after any built-in token events (such as those emitted by `ft-transfer?`) and before the memo in the case of `transfer-memo` and `transfer-many-memo`.
+
+| Event name           | Tuple structure                                                                                       | Description                          |
+|----------------------|-------------------------------------------------------------------------------------------------------|--------------------------------------|
+| `sft_transfer_event` | `{type: "sft_transfer_event", token-id: uint, amount: uint, sender: principal, recipient: principal}` | Emitted when tokens are transferred. |
+| `sft_mint_event`     | `{type: "sft_mint_event", token-id: uint, amount: uint, recipient: principal}`                        | Emitted when new tokens are minted.  |
+| `sft_burn_event`     | `{type: "sft_burn_event", token-id: uint, amount: uint, sender: principal, recipient: principal}`     | Emitted when tokens are burned.      |
```

Comment:
```suggestion
| Event name           | Tuple structure                                                                                       | Description                          |
|----------------------|-------------------------------------------------------------------------------------------------------|--------------------------------------|
| `sft_transfer` | `{type: "sft_transfer", token-id: uint, amount: uint, sender: principal, recipient: principal}` | Emitted when tokens are transferred. |
| `sft_mint`     | `{type: "sft_mint", token-id: uint, amount: uint, recipient: principal}`                        | Emitted when new tokens are minted.  |
| `sft_burn`     | `{type: "sft_burn", token-id: uint, amount: uint, sender: principal}`     | Emitted when tokens are burned.      |
```

I think we there is no need to keep `_event` in events names nor types.
`sft_burn` don't need `recipient`.

Also probably we should keep these events as small as possible, so maybe we should replace `sender` and `recipient` in transfer event with `from` and `to` ?